java_src/foam/dao/JournaledDAO.java

```diff
@@ -29,31 +30,31 @@ public JournaledDAO(DAO delegate, String filename) throws IOException {
    * @returns FObject
    */
   @Override
-  public FObject put(FObject obj) {
+  public FObject put_(X x, FObject obj) {
     journal.put(obj, null);
-    return getDelegate().put(obj);
+    return getDelegate().put_(x, obj);
   }
 
   @Override
-  public FObject remove(FObject obj) {
+  public FObject remove_(X x, FObject obj) {
     Object id = ((AbstractDAO) getDelegate()).getPrimaryKey().get(obj);
     journal.remove(id, null);
-    return getDelegate().remove(obj);
+    return getDelegate().remove_(x, obj);
   }
 
   @Override
-  public FObject find(Object id) {
+  public FObject find_(X x, Object id) {
     return null;
   }
 
   @Override
-  public Sink select(Sink sink, Integer skip, Integer limit, Comparator order, Predicate predicate) {
+  public Sink select_(X x, Sink sink, Integer skip, Integer limit, Comparator order, Predicate predicate) {
     return null;
   }
 
   @Override
-  public void removeAll(Integer skip, Integer limit, Comparator order, Predicate predicate) {
+  public void removeAll_(X x, Integer skip, Integer limit, Comparator order, Predicate predicate) {
     journal.removeAll();
```

Comment:
calling removeAll on journal would mean I would be refactoring a non DAO object. If we want to make this change, I have to change `FileJournal.java` as well.